Automattic/wp-calypso
# Launchpad: Migrate start-writing flow to use the new structure

Related to #86647

## Proposed Changes
Migrate the tasks which are part of the flow `start-writing` 

| TASK | MIGRATED? |
|--------|--------|
| first_post_published |  Previously Migrated |
| setup_blog | THIS PR |
| 'domain_upsell' | Previously Migrated |
| plan_completed | THIS PR |
| blog_launched | THIS PR | 

### Flags Expected State
**production**:                    false
**development**:                   true
**stage**:                         true
**test**:                          true



## Testing Instructions
* Start the flow `start-writing`  (setup/start-writing)
* Choose 'New site' 
* Set a post title and content
* Publish the post
* Wait the flow redirect you to `setup/start-writing/launchpad?siteSlug=[YOUR_SITE]`
* Enable the feature flags via URL and reload
* Check if  there are messages on devtools to all task above
* Use the launchpad (Set values, Edit values, Check if the values are marked when completed)
* Finish the flow.

Requires the feature flags:
* "launchpad/new-task-definition-parser"
* "launchpad/new-task-definition-parser/start-writing"

`flags=launchpad/new-task-definition-parser,launchpad/new-task-definition-parser/start-writing` 

The flags can be enabled appending  the `?flags=` flag on the url (Please always check if the flags= param is still on the url when the system redirect back the user from some "no launchpad" page ).

## Pre-merge Checklist
